diff --git a/node_modules/react-native/.DS_Store b/node_modules/react-native/.DS_Store
new file mode 100644
index 0000000..852087d
Binary files /dev/null and b/node_modules/react-native/.DS_Store differ
diff --git a/node_modules/react-native/Libraries/LogBox/UI/LogBoxInspectorHeader.js b/node_modules/react-native/Libraries/LogBox/UI/LogBoxInspectorHeader.js
index 2987298..9dea7ea 100644
--- a/node_modules/react-native/Libraries/LogBox/UI/LogBoxInspectorHeader.js
+++ b/node_modules/react-native/Libraries/LogBox/UI/LogBoxInspectorHeader.js
@@ -52,7 +52,7 @@ function LogBoxInspectorHeader(props: Props): React.Node {
         <LogBoxInspectorHeaderButton
           disabled={props.total <= 1}
           level={props.level}
-          image={require('./LogBoxImages/chevron-left.png')}
+          // image={require('./LogBoxImages/chevron-left.png')}
           onPress={() => props.onSelectIndex(prevIndex)}
         />
         <View style={styles.title}>
@@ -61,7 +61,7 @@ function LogBoxInspectorHeader(props: Props): React.Node {
         <LogBoxInspectorHeaderButton
           disabled={props.total <= 1}
           level={props.level}
-          image={require('./LogBoxImages/chevron-right.png')}
+          // image={require('./LogBoxImages/chevron-right.png')}
           onPress={() => props.onSelectIndex(nextIndex)}
         />
       </View>
diff --git a/node_modules/react-native/Libraries/LogBox/UI/LogBoxInspectorSourceMapStatus.js b/node_modules/react-native/Libraries/LogBox/UI/LogBoxInspectorSourceMapStatus.js
index 4036783..82f70a9 100644
--- a/node_modules/react-native/Libraries/LogBox/UI/LogBoxInspectorSourceMapStatus.js
+++ b/node_modules/react-native/Libraries/LogBox/UI/LogBoxInspectorSourceMapStatus.js
@@ -71,11 +71,11 @@ function LogBoxInspectorSourceMapStatus(props: Props): React.Node {
   let color;
   switch (props.status) {
     case 'FAILED':
-      image = require('./LogBoxImages/alert-triangle.png');
+      // image = require('./LogBoxImages/alert-triangle.png');
       color = LogBoxStyle.getErrorColor(1);
       break;
     case 'PENDING':
-      image = require('./LogBoxImages/loader.png');
+      // image = require('./LogBoxImages/loader.png');
       color = LogBoxStyle.getWarningColor(1);
       break;
   }
diff --git a/node_modules/react-native/Libraries/LogBox/UI/LogBoxNotification.js b/node_modules/react-native/Libraries/LogBox/UI/LogBoxNotification.js
index 9137616..e4125f5 100644
--- a/node_modules/react-native/Libraries/LogBox/UI/LogBoxNotification.js
+++ b/node_modules/react-native/Libraries/LogBox/UI/LogBoxNotification.js
@@ -105,7 +105,7 @@ function DismissButton(props: $TEMPORARY$object<{onPress: () => void}>) {
         onPress={props.onPress}
         style={dismissStyles.press}>
         <Image
-          source={require('./LogBoxImages/close.png')}
+          // source={require('./LogBoxImages/close.png')}
           style={dismissStyles.image}
         />
       </LogBoxButton>
diff --git a/node_modules/react-native/Libraries/NativeComponent/NativeComponentRegistryUnstable.js b/node_modules/react-native/Libraries/NativeComponent/NativeComponentRegistryUnstable.js
index cebacfc..ded6775 100644
--- a/node_modules/react-native/Libraries/NativeComponent/NativeComponentRegistryUnstable.js
+++ b/node_modules/react-native/Libraries/NativeComponent/NativeComponentRegistryUnstable.js
@@ -3,6 +3,8 @@
  *
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
  */
 
 let componentNameToExists: Map<string, boolean> = new Map();
